### PR TITLE
add skeleton for lsp symbol support

### PR DESCRIPTION
**Pull request by manoj2601**
#2051 

1. added `lsp/symbols` 
2. add `Symbol` in `Request`
3. defined `parseSymbol` function to parse json value to 'Symbol'
4. created object type for `SymbolInformation`.
5. **_TODO_**: implement `processSymbol` function

### CODE REVIEW

File: main/src/ca/uwaterloo/flix/api/lsp/LanguageServer.scala [hunk 1]
```diff
@@ -351,6 +354,15 @@ class LanguageServer(port: Int) extends WebSocketServer(new InetSocketAddress("l
...
   }
 
+
+  /**
+   * Processes a symbol request
+   */
+  private def processSymbol(requestId: String, uri: String, pos: Position): JValue = {
+
+    // TODO implement the function
+    ("id" -> requestId) ~ ("status" -> "success")
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/api/lsp/LanguageServer.scala:364]: Could you start by returning a hardcoded list?

Comment (manoj2601) [reply main/src/ca/uwaterloo/flix/api/lsp/LanguageServer.scala:364]: What should be the return type of this function?

`val result = SymbolProvider.getSymbol(uri, pos) `
 `("id" -> requestId) ~ ("status" -> "success") ~ ("result" -> result.toJSON)`
Here `result` is an hardcoded object of `SymbolInformation`. It is not working and showing this :
Screenshot from 2021-07-22 12-55-52

Comment (manoj2601) [reply main/src/ca/uwaterloo/flix/api/lsp/LanguageServer.scala:364]: > Could you start by returning a hardcoded list?

Why should be there a list? At any location, there can be at most one symbol I guess.

Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/api/lsp/LanguageServer.scala]: https://microsoft.github.io/language-server-protocol/specifications/specification-current/#textDocument_documentSymbol

**Pull request closed without merging**